### PR TITLE
Flip user video

### LINKED ISSUE #3378: Mirror my video
Is it possible to mirror my video when I use video calls?

Other services do this, it's natural to see yourself like if you would look into mirror.

### PR DESCRIPTION
**Pull request by mhlavac**
@RocketChat/core 

Closes #3378

I've noticed that my video was not mirrored. This uses CSS transform to flip my video on my preview + on main video if I am shown. It will not flip the video if screen is shared.

### CODE REVIEW

File: packages/rocketchat-theme/assets/stylesheets/base.less [hunk 1]
```diff
@@ -4294,6 +4294,14 @@ body:not(.is-cordova) {
 			font-weight: bold;
 		}
 	}
+	.video-flip {
+		-moz-transform: scaleX(-1);
+		-o-transform: scaleX(-1);
+		-webkit-transform: scaleX(-1);
+		transform: scaleX(-1);
+		filter: FlipH;
+		-ms-filter: "FlipH";
```
Comment (mhlavac) [inline packages/rocketchat-theme/assets/stylesheets/base.less]: should some prefixer or macro be used insetad?

File: packages/rocketchat-ui/views/app/videoCall/videoCall.coffee [hunk 2]
```diff
@@ -61,6 +61,28 @@ Template.videoCall.helpers
...
+			if remoteItems?.length > 0
+				myself = remoteItems[0].url == localUrl
+
+			if !myself
+				myself = webrtc.localUrl.get() == localUrl
+
+		if !myself && webrtc.remoteItemsById.get()[template.mainVideo.get()]?
+			myself = webrtc.remoteItemsById.get()[template.mainVideo.get()].url == localUrl
+
+		return myself && !WebRTC.getInstanceByRoomId(Session.get('openedRoom')).screenShareEnabled.get()
```
Comment (mhlavac) [inline packages/rocketchat-ui/views/app/videoCall/videoCall.coffee]: Any ideas how to reuse logic from mainVideoUrl. All I need to check is if currently shown `mainVideoUrl == selfVideoUrl`

File: packages/rocketchat-theme/assets/stylesheets/base.less (hunk 1 above)
Comment (rodrigok) [inline packages/rocketchat-theme/assets/stylesheets/base.less]: @mhlavac we are using https://github.com/less/less-plugin-autoprefix so you don't need to prefix theses items

File: packages/rocketchat-ui/views/app/videoCall/videoCall.coffee (hunk 2 above)
Comment (rodrigok) [inline packages/rocketchat-ui/views/app/videoCall/videoCall.coffee]: Maybe you can call the helper like
```html
<video src="{{mainVideoUrl}}" autoplay muted="true" class="{{#if $eq mainVideoUrl selfVideoUrl}}video-flip{{/if}}"></video>
```

What do you think?

Comment (mhlavac) [inline packages/rocketchat-ui/views/app/videoCall/videoCall.coffee]: @rodrigok thank you. I'll try it out

**Comment by graywolf336** [comment]
Hey @mhlavac by chance any update on this?

**Comment by mhlavac** [comment]
@graywolf336 sorry, I didn't move this forward. I hope to get back to it this week and use adjustment you suggested.

**Pull request merged**